docs/documentation/advanced/unkown-fields.fsx

```diff
@@ -0,0 +1,302 @@
+
+We need some specific decoder in order to work with our JSON.
+
+*)
+
+module Decode =
+
+(**
+
+1. We need a decoder which ignore failed decoder
+
+*)
+
+    let ignoreFail (decoder : Decoder<'T>) : Decoder<'T option> =
+        fun path token ->
+            match decoder path token with
+            | Ok x -> Ok(Some x)
+            | Error _ -> Ok None
+
+(**
+
+2. We need a decoder which decodes all the object fields and only keep the valid ones
+
+*)
+
+    let keyValueOptions (decoder: Decoder<'a option>) : Decoder<(string * 'a) list> =
+        decoder
+        |> Decode.keyValuePairs
+        |> Decode.map (
+            List.collect
+                (fun (key, aOpt) ->
+                    match aOpt with
+                    | Some a -> [ key, a ]
+                    | None -> [])
+        )
+
+(**
+
+## Define our domain types
+
+First, we need a type to represent the time field.
+*)
+
+
+type Ts =
+    Ts of System.DateTime
+
+module Ts =
+
+    let decoder : Decoder<Ts> =
+        Decode.datetime
+        |> Decode.map Ts
+
+(**
+
+Then a type to represent a valid `Rate` field.
+
+*)
+
+type RateObject =
+    RateObject of decimal
+
+module RateObject =
+
+    let decoder : Decoder<RateObject> =
+        Decode.field "rate" Decode.decimal
+        |> Decode.map RateObject
+
+(**
+
+
+Now, we need a type to store all the information associated to a rate.
+
+This type contains the name of the 2 currencies and the rate.
+
+*)
+
+type Rate =
+    {
+        SourceCurrency : string
+        TargetCurrency : string
+        Rate : decimal
+    }
+
+(**
+
+We don't have a `Decoder<Rate>` because the informations required to build a `Rate` are not stored in a standard object.
+
+Indeed, they are coming from both the field name and the associated value.
+
+In order, to work with this JSON we are going to directly works on all the field of the JSON.
+
+*)
+
+module Rates =
+
+    let decoder : Decoder<Rate list> =
+        // 1. We retrieve all the valid RateObject fields and their associated name
+        Decode.keyValueOptions (Decode.ignoreFail RateObject.decoder)
+        // 2. Now that we have all the potential valid Rate fields
+        // We need to verify if they have a valid name
+        |> Decode.andThen (fun rateObjects ->
+            rateObjects
+            |> List.map (fun (fieldName, (RateObject value)) ->
+                // We consider the fieldName valid if it contains a `_`
+                // The format is [sourceCurrency]_[targetCurrency]
+                let segments = fieldName.Split('_')
+
+                // If the fieldName is invalid
+                // Returns None, this will allow us to filter invalid fields without failing
+                if segments.Length <> 2 then
+                    None
+                else
+                    // The fieldName is valid, we can build the Rate record
+                    let sourceCurrency = segments[0]
+                    let targetCurrency = segments[1]
```

Comment:
I think this would get rid of like more than a half lines of the code:
```fsharp
    match fieldName.Split('_') with
    | [| b; q |] -> 
        Some { BaseCurrency = b; QuoteCurrency = q; Rate = rate }
    | _ -> None
```

Reply:
You are right, I wrote it like that originally and then switched the syntax. 

I am not sure why, I will revisit this section